src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeCustomAttributeData.cs

```diff
@@ -273,6 +273,13 @@ private RuntimeCustomAttributeData(RuntimeModule scope, MetadataToken caCtorToke
             m_scope = scope;
             m_ctor = (RuntimeConstructorInfo)RuntimeType.GetMethodBase(scope, caCtorToken)!;
 
+            if (m_ctor!.DeclaringType!.IsGenericType)
```

Comment:
nit: I don't think the `!` is needed for `m_ctor`

Reply:
It appears this is optimized for the non-generic case, since in the generic case we re-set `m_ctor`. However, without measuring perf, the current approach might be better than getting the "is generic" information from `scope.ResolveType(...).IsGenericType` and only setting `m_ctor` once.

Reply:
Correct, this is optimized for the non-generic case, and is probably non-ideal for the generic case. OTOH, the non-generic case is the perf critical case, so this should be a pretty good result.